pkg/lca/ibu_test.go

```diff
@@ -0,0 +1,753 @@
+package lca
+
+import (
+	"fmt"
+	"testing"
+
+	lcav1alpha1 "github.com/openshift-kni/lifecycle-agent/api/v1alpha1"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+)
+
+var (
+	ibulGvk = schema.GroupVersionKind{
+		Group:   "lca.openshift.io",
+		Version: "v1alpha1",
+		Kind:    "ImageBasedUpgrade",
+	}
+)
+
+func TestImageBasedUpgradeWithOptions(t *testing.T) {
+	testSettings := buildTestClientWithDummyObject()
+	testBuilder, _ := PullImageBasedUpgrade(testSettings)
+	testBuilder = testBuilder.WithOptions(
+		func(builder *ImageBasedUpgradeBuilder) (*ImageBasedUpgradeBuilder, error) {
+			return builder, nil
+		})
+
+	assert.Equal(t, "", testBuilder.errorMsg)
+
+	testBuilder = testBuilder.WithOptions(
+		func(builder *ImageBasedUpgradeBuilder) (*ImageBasedUpgradeBuilder, error) {
+			return builder, fmt.Errorf("error")
+		})
+	assert.Equal(t, "error", testBuilder.errorMsg)
+}
+
+func buildTestClientWithDummyObject() *clients.Settings {
```

Comment:
Please move all private function to the end of the file

Reply:
Done.